### ISSUE TITLE
tf.keras load_model fails to load a model created with Keras

### ISSUE THREAD

**Issue by cmwolverton** [tensorflow/tensorflow#25200]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Win7 same issue on Ubuntu
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):   b'v1.11.0-rc2-4-gc19e29306c' 1.11.0
- Python version:  Python 3.6.6
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: CUDA Toolkit 9.0 (Sept 2017) cuDNN v7.0.5 (Dec 5, 2017)
- GPU model and memory: device: 0, name: Quadro M4000, compute capability: 5.2 memory_limit: 7801630311
                                            device: 1, name: Quadro M4000, compute capability: 5.2 memory_limit: 7801630311
- Keras version: 2.2.4

I am trying to load a model saved with Keras (not tf.keras) to create an Estimator.    The Keras model is a slightly modified version of Inception_V3 from the Keras model library.  (The modification is that the final layer was replaced with a dense layer and sigmoid to use the model for regression.)  

A co-worker was able to load a standard (unmodified) Keras model with tf.keras.  So I believe the intended behavior is that all Keras created models are loadable.  Please let me know if I have a misunderstanding about the intended functionality.

**Describe the current behavior**
Notebook1: Save the Keras model like this (abbreviated steps):

from keras.applications.inception_v3 import InceptionV3
base_model =InceptionV3(include_top=True, weights=viewClassWeightsFile, input_shape=input_shape, classes=nb_classes)
base_model.layers.pop() 
base_model.outputs = [base_model.layers[-1].output]
base_model.layers[-1].outbound_nodes=[]
output = base_model.get_layer('avg_pool').output
output = Dense(activation="sigmoid", units=num_regression_targets)(output)
new_model = Model(base_model.input, output)

new_model.save(modelSaveFile) 

Notebook2: Load the Keras model with tf.keras
new_model = tf.keras.models.load_model(modelSaveFile)

AttributeError: module 'tensorflow.python.keras.backend' has no attribute 'slice'

**Describe the expected behavior**

I expected that tf.keras.models.load_model() should be able to load a model created witih Keras.  

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

Here is full traceback:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-0e6b1a78cac7> in <module>
      6 modelSaveFile = modelDir+'\\model_singlePt_mix6_mlx100_set2.h5'
      7 
----> 8 base_model = keras.models.load_model(modelSaveFile)
      9 
     10 

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\engine\saving.py in load_model(filepath, custom_objects, compile)
    228       raise ValueError('No model found in config file.')
    229     model_config = json.loads(model_config.decode('utf-8'))
--> 230     model = model_from_config(model_config, custom_objects=custom_objects)
    231 
    232     # set weights

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\engine\saving.py in model_from_config(config, custom_objects)
    308                     '`Sequential.from_config(config)`?')
    309   from tensorflow.python.keras.layers import deserialize  # pylint: disable=g-import-not-at-top
--> 310   return deserialize(config, custom_objects=custom_objects)
    311 
    312 

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\layers\serialization.py in deserialize(config, custom_objects)
     62       module_objects=globs,
     63       custom_objects=custom_objects,
---> 64       printable_module_name='layer')

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\utils\generic_utils.py in deserialize_keras_object(identifier, module_objects, custom_objects, printable_module_name)
    171             custom_objects=dict(
    172                 list(_GLOBAL_CUSTOM_OBJECTS.items()) +
--> 173                 list(custom_objects.items())))
    174       with CustomObjectScope(custom_objects):
    175         return cls.from_config(config['config'])

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\engine\network.py in from_config(cls, config, custom_objects)
   1300         if layer in unprocessed_nodes:
   1301           for node_data in unprocessed_nodes.pop(layer):
-> 1302             process_node(layer, node_data)
   1303 
   1304     name = config.get('name')

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\engine\network.py in process_node(layer, node_data)
   1258       if input_tensors:
   1259         if len(input_tensors) == 1:
-> 1260           layer(input_tensors[0], **kwargs)
   1261         else:
   1262           layer(input_tensors, **kwargs)

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\engine\base_layer.py in __call__(self, inputs, *args, **kwargs)
    767 
    768       if not in_deferred_mode:
--> 769         outputs = self.call(inputs, *args, **kwargs)
    770         if outputs is None:
    771           raise ValueError('A layer\'s `call` method should return a Tensor '

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\layers\core.py in call(self, inputs, mask)
    715       arguments['mask'] = mask
    716     print(self.function.__name__) # added by Cindy
--> 717     return self.function(inputs, **arguments)
    718 
    719   def compute_mask(self, inputs, mask=None):

c:\users\wolvci10\appdata\local\continuum\anaconda3\envs\tf_1p9\lib\site-packages\tensorflow\python\keras\layers\core.py in get_slice(data, i, parts)
    193         stride = K.concatenate([step, input_shape * 0], axis=0)
    194         start = stride * i
--> 195         return K.slice(data, start, size)
    196 
    197     # Relocate the model definition under CPU device scope if needed

AttributeError: module 'tensorflow.python.keras.backend' has no attribute 'slice'

**Comment by gndowns** [comment]
@jvishnuvardhan This does feel like a performance bug though? Are `tf.keras.models.load_model` and `keras.models.Sequential.save_model` not meant to be compatible?

**Comment by danvargg** [comment]
Having the same issue.

**Comment by jvishnuvardhan** [comment]
@danvargg Please post a new issue with details of your issue and a standalone code to reproduce the issue. Thanks!

**Comment by gndowns** [comment]
@jvishnuvardhan why not address the issue & code already linked here? Many people are having the same problem with similar code.

**Comment by gndowns** [comment]
This seems like a major backwards-compatability issue. And if it's not meant to be supported can you please say so?

**Comment by jvishnuvardhan** [comment]
@danvargg I tried it in `TF2.x` and the [gist is here](https://colab.sandbox.google.com/gist/jvishnuvardhan/43e5c9782daaa067ebbef723af9dfdd0/untitled756.ipynb). Thanks!

**Comment by gndowns** [comment]
@jvishnuvardhan Your gist does not address the problem at all, as it imports `keras` from `tensorflow`. The issue is describing an icompatability between `tf.keras` and the original standalone `keras` module.

**Comment by danvargg** [comment]
I did find a workaround (which I don't like too much). My issue is `modelcheckpoint` not being compatible with loading the model in `TF2.x`.

I managed to do so by saving the callback as `.hdf5` (which it seems to work with `TF2.x`) and then doing `model.save('model.h5')` on the `.hdf5`.

After that, I was able to load the `.h5` model and convert it with `tfcoreml`. I'm currently testing if the predictions of all these models produce the same results to ensure nothing changed between saves.

I really don't like doing workarounds like this, specially for models that will go to prod.

@jvishnuvardhan, is there a better approach to load and convert models saved with `modelcheckpoint`?
